### PR TITLE
[#290] Screen on for UI tests

### PR DESCRIPTION
**Pull request by ccjernigan**
This improves the reliability of UI tests, especially on physical devices, by helping ensure the screen is on.

To verify this is working, try running the UI and app tests on a physical device which has the screen turned off at the start.  You can also try turning off the screen while the tests are running.  In each case, the assertion error message will be much more clear about why the test failed and the tests will fail much more quickly.

This code review checklist is intended to serve as a starting point for the author and reviewer, although it may not be appropriate for all types of changes (e.g. fixing a spelling typo in documentation).  For more in-depth discussion of how we think about code review, please see [Code Review Guidelines](../blob/main/docs/CODE_REVIEW_GUIDELINES.md).

# Author

- [x] Self-review: Did you review your own code in GitHub's web interface? _Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs._
- [ ] Automated tests: Did you add appropriate automated tests for any code changes?
- [ ] Manual tests: Did you update the [manual tests](../blob/main/docs/testing/manual_testing) as appropriate? _While we aim for automated testing of the application, some aspects require manual testing. If you had to manually test something during development of this pull request, write those steps down._
- [ ] Code coverage: Did you check the code coverage report for the automated tests?  _While we are not looking for perfect coverage, the tool can point out potential cases that have been missed._ Code Coverage can be run with: `./gradlew connectedCheck -PIS_COVERAGE_ENABLED=true`
- [ ] Documentation: Did you update documentation as appropriate? (e.g [README.md](../blob/main/README.md), etc.)
- [ ] Run the app: Did you run the app and try the changes?
- [ ] Screenshots: Did you provide before and after UI screenshots in the description of this pull request?  _This is only applicable for changes that modify the UI._
- [x] Rebase and squash: Did you pull in the latest changes from the main branch and squash your commits before assigning a reviewer? _Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit._

# Reviewer

- [x] Checklist review: Did you go through the code with the [Code Review Guidelines](../blob/main/docs/CODE_REVIEW_GUIDELINES.md) checklist?
- [x] Ad hoc review: Did you perform an ad hoc review?  _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
- [x] Automated tests: Did you review the automated tests?
- [ ] Manual tests: Did you review the manual tests?
- [ ] How is Code Coverage affected by this PR? _We encourage you to compare coverage before and after changes and when possible, leave it in a better place._
- [ ] Documentation: Did you review Docs, [README.md](../blob/main/README.md), and [Architecture.md](../blob/main/docs/Architecture.md) as appropriate?
- [x] Run the app: Did you run the app and try the changes? _While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data. Perform this step last, after verifying the code changes are safe to run locally._

### CODE REVIEW

**Review by HonzaR — APPROVED** [review]
It's very useful improvement and it also saves some of our CI run time.

File: test-lib/src/main/kotlin/co/electriccoin/zcash/test/ZcashUiTestRunner.kt [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+class ZcashUiTestRunner : AndroidJUnitRunner() {
+    private lateinit var wakeLock: PowerManager.WakeLock
+
+    override fun onCreate(arguments: Bundle?) {
+        super.onCreate(arguments)
+
+        // There is no alternative to this deprecated API.  The suggestion of a view to keep the screen
+        // on won't work well for our tests.
+        @Suppress("DEPRECATION")
+        wakeLock = (
```
[the remark this answers is not in the document]
Comment (HonzaR) [reply test-lib/src/main/kotlin/co/electriccoin/zcash/test/ZcashUiTestRunner.kt:18]: Yep, that's right. I just did a quick research on this topic and I haven't found anything useful for us.. So lets go with the wakeLock solution as log as we can or a better solution comes up.

[the remark this answers is not in the document]
Comment (ccjernigan) [reply test-lib/src/main/kotlin/co/electriccoin/zcash/test/ZcashUiTestRunner.kt:18]: Test runner onCreate() is called each time TestOrchestrator relaunches the test process.  So I added the flag `ON_AFTER_RELEASE` to poke the idle timeout timer. The delay between relaunches should be smaller than even very brief screen off timeouts.

[the remark this answers is not in the document]
Comment (HonzaR) [reply test-lib/src/main/kotlin/co/electriccoin/zcash/test/ZcashUiTestRunner.kt]: Checking the `ON_AFTER_RELEASE`, which is good to have, I bumped into this flag `ACQUIRE_CAUSES_WAKEUP` [doc](https://developer.android.com/reference/android/os/PowerManager#ACQUIRE_CAUSES_WAKEUP), which could be also useful to set up if we repeatedly ask for WakeLock.

[the remark this answers is not in the document]
Comment (ccjernigan) [reply test-lib/src/main/kotlin/co/electriccoin/zcash/test/ZcashUiTestRunner.kt]: I think the KeyGuard is going to get in the way of `ACQUIRE_CAUSES_WAKEUP`.  Even if the screen turns on, the KeyGuard will prevent our test Activity from being in the foreground.

**Pull request merged**